src/Service.Tests/SqlTests/GraphQLSupportedTypesTests/GraphQLSupportedTypesTestsBase.cs

```diff
@@ -86,6 +87,110 @@ public async Task QueryTypeColumn(string type, int id)
+        [DataRow(FLOAT_TYPE, "gt", "-9.2", "-9.2", ">")]
+        [DataRow(FLOAT_TYPE, "gte", "-9.2", "-9.2", ">=")]
+        [DataRow(FLOAT_TYPE, "lt", ".33", "0.33", "<")]
+        [DataRow(FLOAT_TYPE, "lte", ".33", "0.33", "<=")]
+        [DataRow(FLOAT_TYPE, "neq", "-9.2", "-9.2", "!=")]
+        [DataRow(FLOAT_TYPE, "eq", "-9.2", "-9.2", "=")]
+        [DataRow(DECIMAL_TYPE, "gt", "-9.292929", "-9.292929", " > ")]
+        [DataRow(DECIMAL_TYPE, "gte", "-9.292929", "-9.292929", " >= ")]
+        [DataRow(DECIMAL_TYPE, "lt", "0.333333", "0.333333", "<")]
+        [DataRow(DECIMAL_TYPE, "lte", "0.333333", "0.333333", " <= ")]
+        [DataRow(DECIMAL_TYPE, "neq", "0.0", "0.0", "!=")]
+        [DataRow(DECIMAL_TYPE, "eq", "-9.292929", "-9.292929", "=")]
+        [DataRow(BOOLEAN_TYPE, "neq", "\'false\'", "false", "!=")]
+        [DataRow(BOOLEAN_TYPE, "eq", "\'false\'", "false", "=")]
+        public async Task QueryTypeColumnFilterAndOrderBy(string type, string filterOperator, string sqlValue, string gqlValue, string queryOperator)
+        {
+            if (!IsSupportedType(type))
+            {
+                Assert.Inconclusive("Type not supported");
+            }
+
+            string field = $"{type.ToLowerInvariant()}_types";
+            string graphQLQueryName = "supportedTypes";
+            string gqlQuery = @"{
+                supportedTypes(first: 100 orderBy: { " + field + ": ASC } filter: { " + field + ": {" + filterOperator + ": " + gqlValue + @"} }) {
+                    items {
+                        " + field + @"
+                    }
+                }
+            }";
+
+            string dbQuery = MakeQueryOnTypeTable(new List<string> { field }, filterValue: sqlValue, filterOperator: queryOperator, filterField: field, orderBy: field, limit: "100");
+
+            JsonElement actual = await ExecuteGraphQLRequestAsync(gqlQuery, graphQLQueryName, isAuthenticated: false);
+            string expected = await GetDatabaseResultAsync(dbQuery);
+
+            PerformTestEqualsForExtendedTypes(type, expected, actual.GetProperty("items").ToString());
+        }
+
+        [DataTestMethod]
+        [DataRow(DATETIME_TYPE, "gt", "\'1999-01-08\'", "\"1999-01-08\"", " > ")]
+        [DataRow(DATETIME_TYPE, "gte", "\'1999-01-08\'", "\"1999-01-08\"", " >= ")]
+        [DataRow(DATETIME_TYPE, "lt", "\'0001-01-01\'", "\"0001-01-01\"", " < ")]
+        [DataRow(DATETIME_TYPE, "lte", "\'0001-01-01\'", "\"0001-01-01\"", " <= ")]
+        [DataRow(DATETIME_TYPE, "neq", "\'0001-01-01\'", "\"0001-01-01\"", "!=")]
+        [DataRow(DATETIME_TYPE, "eq", "\'0001-01-01\'", "\"0001-01-01T01:01:01\"", "=")]
+        [DataRow(DATETIME_TYPE, "gt", "\'1999-01-08 10:23:00\'", "\"1999-01-08 10:23:00\"", " > ")]
+        [DataRow(DATETIME_TYPE, "gte", "\'1999-01-08 10:23:00\'", "\"1999-01-08 10:23:00\"", " >= ")]
+        [DataRow(DATETIME_TYPE, "lt", "\'9998-12-31 23:59:59\'", "\"9998-12-31 23:59:59\"", " < ")]
```

Comment:
our implementation of GQL will not be able to handle dates with year 9999 since it goes over its upper limit. This should work now locally and in the pipeline, ill add a comment explaining the choice of 9998.